src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -307,17 +307,23 @@ private Host handleDuplicateHost(Set<String> duplicateInstanceIds, Host host1, H
 
   public static void populateHostFieldsFromHbi(
       Host host, InventoryHostFacts inventoryHostFacts, NormalizedFacts normalizedFacts) {
-    if (inventoryHostFacts.getInventoryId() != null) {
-      host.setInventoryId(inventoryHostFacts.getInventoryId().toString());
-      // We assume that the instance ID for any given HBI host record is the inventory ID; compare
-      // to an OpenShift Cluster from Prometheus data, where we use the cluster ID.
-      if (host.getInstanceId() == null) {
-        // Don't overwrite the instanceId if already set, since that would cause potential
-        // duplicates in the serviceInstances map in AccountServiceInventory.
+    // Don't overwrite the instanceId if already set, since that would cause potential
+    // duplicates in the serviceInstances map in AccountServiceInventory.
+    if (host.getInstanceId() == null) {
+      if (inventoryHostFacts.getProviderId() != null) {
+        // will use the provider ID from HBI if sets:
+        host.setInstanceId(inventoryHostFacts.getProviderId());
+      } else if (inventoryHostFacts.getInventoryId() != null) {
```

Comment:
do we want to keep the previous functionality?

Reply:
@kahowell Does the previous functionality need to be maintained as a fallback?

Reply:
ping @kahowell

Reply:
@Sgitario yes, we need to fallback to previous logic.